src/service/file_manager/file_manager.service.ts

```diff
@@ -5,12 +5,25 @@ import { StorageService } from '../storage/storage.service';
 import { Folder } from '../../model/folder';
 import { FolderFactory } from './folder.factory';
 import { File, User } from '../../model';
+import { FileFactory } from './file.factory';
+import { DeleteFileRequest, DeleteFolderRequest, UploadFileRequest } from 'src/interface/apiRequest';
+import { DeleteFileResponse, DeleteFolderResponse, UploadFileResponse } from 'src/interface/apiResponse';
+import { checkForUpwardDir } from 'src/shared/util/file.utils';
+import { ApplicationError } from 'src/shared/error/applicationError';
 
 @Injectable()
 export class FileManagerService {
   constructor(private readonly storageService: StorageService) {}
 
+  private static getUserLocation(user: User, location: string, isFile?: boolean): string {
+    const prefix = `organization/${user.id}/user/${user.id}`;
+    const wLocation = location.endsWith('/') || isFile ? location : `${location}/`;
+    return path.posix.join(prefix, wLocation);
+  }
+
   public async createFolder(user: User, location: string, name: string): Promise<Folder> {
+    if (checkForUpwardDir(location)) throw new UpwardDirectoryError('Trying to reach a directory above root');
```

Comment:
it's better not to move it to utils. Let's have a method called `ensureLocationCanBeUsed`. It will a method of `FileManagerService`. And we will call it from our controller. Then, method `createFolder` will be responsible only for creating folder.